flutter-engine/src/assets.rs

```diff
@@ -0,0 +1,22 @@
+use std::path::PathBuf;
+
+pub fn asset_dir() -> PathBuf {
+    if let Some(proj_dir) = std::env::var("CARGO_MANIFEST_DIR").ok() {
+        return PathBuf::from(&proj_dir)
+            .join("target")
+            .join("flutter_assets");
+    }
+    platform_asset_dir()
+}
+
+#[cfg(target_os = "linux")]
+fn platform_asset_dir() -> PathBuf {
+    std::env::current_exe()
```

Comment:
This path is rather random?

Is there any reason for this to be part of `flutter-engine` itself?

Reply:
this is the path that works for appimages built with `cargo-flutter`, and I suspect other package formats will work the same. `/usr/bin/app` -> `../share/flutter_assets`. If we suggest `cargo-flutter` as the way to build `flutter-rs` projects it makes sense to have some kind support for loading them.

```rust
use flutter_engine::assets::asset_dir;
use flutter_winit::FlutterWindow;
use glutin::window::WindowBuilder;

fn main() {
    env_logger::init();
    let window = WindowBuilder::new();
    let flutter = FlutterWindow::new(window).unwrap();
    let flutter = flutter.with_resource_context().unwrap();
    flutter.start_engine(&asset_dir(), &[]).unwrap();
    flutter.run();
}
```